hathor/transaction/aux_pow.py

```diff
@@ -19,7 +19,8 @@
 logger = get_logger()
 
 
-MAX_MERKLE_PATH_LENGTH: int = 12
+OLD_MAX_MERKLE_PATH_LENGTH: int = 12
+NEW_MAX_MERKLE_PATH_LENGTH: int = 15
```

Comment:
The reason I chose `15` is because I ran a query on our AWS logs for the last 12 weeks, and in that time we had:

- 1591 instances of `merkle_path too long: 13 > 12`
- 757 instances of `merkle_path too long` (from before the logging was improved)
- 14 instances of `merkle_path too long: 15 > 12`
- 1 instance of `merkle_path too long: 14 > 12`

This of course includes only miners connected to our own nodes. We can still change this number, but I will detail this reasoning further on the RFC.

Reply:
For the testnet, I think 15 (or mostly whatever increase), works, mostly because we'll be able to test. We should also prepare to be able to merge mine, at least against a "dummy chain" (we can use zero'd or random bytes to build dummy data). But for the mainnet, I think we should:

1. at least estimate what parameter would fit the chains most commonly merged-mined with (BTC/DGC/BSV/BCH/BTG), some of those have unlimited blocks, so the theoretical limit of those doesn't matter, but at least scanning the blocks with most transactions on those chains
2. and maybe most importantly, getting feedback from the main miners, since they are directly affected by it

The most pressing counterpoint to a high limit is the block size, which increases with `merkle_path_len * 32 bytes`, the `merkle_path_len` doesn't increase proportionally with the max allowed limit though, the max allowed limit only expands the distribution of allowed values. I'm not sure what's a good way to measure this, I think it's probably not harmful to err a little bit to a higher limit.

Reply:
Agreed. I'm keeping it as 15 then, for testnet. We can do that investigation to chose a better value for mainnet before configuring its activation process.

Reply:
Changed to 20 in https://github.com/HathorNetwork/hathor-core/pull/910/commits/8aa74cba4e8edb7108bf64d45c7af7ba7adca292, as it's most likely the number we'll also use on mainnet.

Reply:
Perfect! I guess you can mark this conversation as resolved.